File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int m = 0;
        int n = 0;
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i = 0; i<nums.length; i++){
            map.put(nums[i],i);
        }
        for(int j = 0; j<nums.length; j++){
            int rest = target - nums[j];
            if(map.containsKey(rest)){
                if(map.get(rest) == j){
                    continue;
                }
                else{
                    m = j;
                    n = map.get(rest);
                    break;
                }
            }
        }
        return new int[]{m,n};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #one pass solution
        numMap = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in numMap:
                return [i,numMap[diff]]
            numMap[nums[i]] = i
                
    #Two pass solution
    def twoSumPass(self, nums,target):
        numMap = {}
        for i in range(len(nums)):
            numMap[nums[i]] = i
        
        for i in range(len(nums)):
            number = target - nums[i]
            if number in numMap and numMap[number] != i:
                return [i, numMap[number]]
        return []
        